MetaMask/metamask-extension
# feat(migration): enable token auto-detection when basic functionality is on

## **Description**

Adds a migration that enables token auto-detection if the basic functionality toggle is on.

It also removes an unused property `showTokenAutodetectModalOnUpgrade` from the app metadata controller.

It is versioned 125.1 because it's planned to be cherry picked to release 12.3, and that's the subsequent migration version in that branch.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/26406?quickstart=1)

## **Related issues**

## **Manual testing steps**

- Install an older build like tag `v12.1.0`
- In settings, disable token auto detection but keep basic functionality on
- Upgrade to this build, reload extension
  - should see `Running migration 125.1...` in console
- Verify token autodetection is enabled in settings
- Download state from advanced settings
- Verify state does not contain `showTokenAutodetectModalOnUpgrade`

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
